webapp/tests/karma/ts/services/db-sync.service.spec.ts

```diff
@@ -770,6 +786,71 @@ describe('DBSync service', () => {
+        isOnlineOnly.returns(false);
+        hasAuth.resolves(true);
+
+        await service.sync();
+        expectSyncCall(1);
+        expect(store.dispatch.callCount).to.equal(0);
+      });
+
+      it('displays a snackbar when the sync begins and when it succeeds', async () => {
+        isOnlineOnly.returns(false);
+        hasAuth.resolves(true);
+
+        await service.sync(true);
+        expectSyncCall(1);
+        expect(store.dispatch.callCount).to.equal(2);
+        expect(store.dispatch.args[0][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[0][0].payload.message).to.equal('sync.status.in_progress');
+        expect(store.dispatch.args[1][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[1][0].payload.message).to.equal('sync.status.not_required');
+      });
+
+      it('displays a snackbar when the sync fails', async () => {
+        isOnlineOnly.returns(false);
+        hasAuth.resolves(true);
+
+        replicationResultTo = replicationResultFrom = Promise.reject('error');
+        await service.sync(true);
+        expectSyncCall(1);
+        expect(store.dispatch.callCount).to.equal(2);
+        expect(store.dispatch.args[0][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[0][0].payload.message).to.equal('sync.status.in_progress');
+        expect(store.dispatch.args[1][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[1][0].payload.message).to.equal('sync.feedback.failure.unknown');
+      });
+
+      it('allows retrying from the snackbar when the sync fails', fakeAsync(async () => {
+        isOnlineOnly.returns(false);
+        hasAuth.resolves(true);
+
+        replicationResultTo = replicationResultFrom = Promise.reject('error');
+        await service.sync(true);
+        expectSyncCall(1);
+        expect(store.dispatch.callCount).to.equal(2);
+        expect(store.dispatch.args[0][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[0][0].payload.message).to.equal('sync.status.in_progress');
+        expect(store.dispatch.args[1][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[1][0].payload.message).to.equal('sync.feedback.failure.unknown');
+
+        replicationResultTo = replicationResultFrom = Promise.resolve();
+        await store.dispatch.args[1][0].payload.action.onClick();
+        tick(1000); // trigger the hide snackbar timeout
+        expectSyncCall(2);
+        expect(store.dispatch.callCount).to.equal(4);
+        expect(store.dispatch.args[2][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[2][0].payload.message).to.equal('sync.status.in_progress');
+        expect(store.dispatch.args[3][0].type).to.equal('SET_SNACKBAR_CONTENT');
+        expect(store.dispatch.args[3][0].payload.message).to.equal('sync.status.not_required');
+        discardPeriodicTasks();
```

Comment:
Because `tick` is not required anymore, you can remove the `fakeAsync` wrapper. This means you don't need to `beforeEach(() => clock.restore());` and you don't need `discardPeriodicTasks`.